### PR TITLE
[node-forge] Added pkcs12 typings

### PR DESCRIPTION
**Pull request by flynetworks**
Please fill in this template.

- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `tsc` without errors.
- [ ] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/digitalbazaar/forge#pkcs12
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: node-forge/index.d.ts [hunk 1]
```diff
@@ -234,4 +236,39 @@ declare module "node-forge" {
...
+            getBags: (filter: BagsFilter) => {
+                [key: string]: Bag[];
+                localKeyId?: Bag[];
+                friendlyName?: Bag[];
+            };
+            getBagsByFriendlyName: (fiendlyName: string, bagType: string) => Bag[]
+            getBagsByLocalKeyId: (localKeyId: string, bagType: string) => Bag[]
+        }
+
+        function pkcs12FromAsn1(obj:any, strictOrPassword: boolean|string, password?: string) : Pkcs12Pfx;
```
Comment (westy92) [inline node-forge/index.d.ts]: `strictOrPassword` should be an optional parameter

Comment (westy92) [inline node-forge/index.d.ts]: It might be best to use function overloading here instead of providing the single function to handle multiple situations.

**Review by westy92 — CHANGES_REQUESTED** [review]

File: node-forge/index.d.ts (hunk 1 above)
Comment (flynetworks) [inline node-forge/index.d.ts]: Yes definitely right. Sorry this solution i hadn't in mind :(

**Review by westy92 — APPROVED** [review]

**Comment by westy92** [comment]
Looks good, thanks!

**Comment by flynetworks** [comment]
You're welcome :-) Thanks a lot for the fast merge.

**Comment by flynetworks** [comment]
Can you please merge this pull request?

**Pull request closed without merging**